packages/panna-sdk/src/core/docs/utils.md

```diff
@@ -0,0 +1,342 @@
+};
+
+// Use in transactions
+const transaction = prepareTransaction({
+  client,
+  chain: lisk,
+  to: recipientAddress,
+  value: toWei('1.5') // Send 1.5 tokens
+});
+```
+
+### Custom Decimal Support
+
+```ts
+// For tokens with different decimal places (like USDC with 6 decimals)
+function toTokenUnits(amount: string, decimals: number): bigint {
+  const factor = BigInt(10 ** decimals);
+  const [whole, decimal = '0'] = amount.split('.');
+  const paddedDecimal = decimal.padEnd(decimals, '0').slice(0, decimals);
+  return BigInt(whole) * factor + BigInt(paddedDecimal);
+}
+
+const usdcAmount = toTokenUnits('100.50', 6); // 100500000n (6 decimals)
+```
+
+## 3. Activity Tracking
+
+```ts
+import { getActivitiesByAddress } from 'panna-sdk';
+
+// Get user's transaction history
+const activities = await getActivitiesByAddress({
+  client,
+  address: userAddress,
+  limit: 20,
+  offset: 0 // For pagination
+});
+
+// Process and filter activities
+const sentTransactions = activities.filter(
+  (activity) => activity.type === 'send'
+);
+const failedTransactions = activities.filter(
+  (activity) => activity.status === 'failed'
+);
+
+// Format for display
+function formatTransactionHistory(activities: Activity[]) {
+  return activities.map((activity) => ({
+    date: new Date(activity.timestamp * 1000).toLocaleDateString(),
+    type: activity.type,
+    amount: (Number(BigInt(activity.value)) / 1e18).toString(),
+    counterparty: activity.type === 'receive' ? activity.from : activity.to,
+    status: activity.status,
+    explorerUrl: `https://blockscout.lisk.com/tx/${activity.hash}`
+  }));
+}
+```
+
+## 4. NFT Collections
+
+```ts
+import { getCollectiblesByAddress } from 'panna-sdk';
+
+// Get user's NFT collection
+const nfts = await getCollectiblesByAddress({
+  client,
+  address: userAddress
+});
+
+// Group NFTs by collection
+const groupedNFTs = nfts.reduce(
+  (groups, nft) => {
+    const collection = nft.collection;
+    if (!groups[collection]) groups[collection] = [];
+    groups[collection].push(nft);
+    return groups;
+  },
+  {} as Record<string, typeof nfts>
+);
+
+// Display collection summary
+Object.entries(groupedNFTs).forEach(([collection, nfts]) => {
+  console.log(`${collection}: ${nfts.length} NFTs`);
+});
+```
+
+## 5. Validation & Input Handling
+
+```ts
+import { isValidAddress } from 'panna-sdk';
+
+// Address validation
+const address = '0x742d35Cc6635C0532925a3b8D42f3C2544a3F97e';
+const isValid = isValidAddress(address);
+
+// Form validation helper
+function validateTransactionInput(
+  recipient: string,
+  amount: string,
+  balance?: bigint
+) {
+  const errors: string[] = [];
+
+  // Address validation
+  if (!recipient || !isValidAddress(recipient.trim())) {
+    errors.push('Invalid recipient address');
+  }
+
+  // Amount validation
+  const numAmount = parseFloat(amount);
+  if (!amount || numAmount <= 0) {
+    errors.push('Amount must be greater than 0');
+  }
```

Comment:
I think, we should be able to post with 0 value as well but not negative.

Reply:
Done.